### PR TITLE
Feature/null effort

### PR DESCRIPTION
**Pull request by graham-willis-druid**
Null effort and null date constraints

### CODE REVIEW

File: src/main/java/uk/gov/defra/datareturns/validation/activities/ActivityValidator.java [hunk 1]
```diff
@@ -79,17 +82,30 @@ private boolean checkUniqueRiverPerSubmission(final Activity activity, final Con
...
      * @return true if valid, false otherwise
      */
     private boolean checkDays(final Activity activity, final ConstraintValidatorContext context) {
         boolean valid = checkDaysMandatoryRelease(activity, context) && checkDaysOther(activity, context);
-        if (valid && activity.getDaysFishedWithMandatoryRelease() < 1 && activity.getDaysFishedOther() < 1) {
-            valid = handleError(context, "DAYS_FISHED_NOT_GREATER_THAN_ZERO", ConstraintValidatorContext.ConstraintViolationBuilder::addBeanNode);
+
+        final Optional<Submission> sub = Optional.ofNullable(activity.getSubmission());
+
+        if (sub.isPresent()) {
```
Comment (sgardnerdell-ea) [inline src/main/java/uk/gov/defra/datareturns/validation/activities/ActivityValidator.java]: It is not possible to have an Activity without as associated Submission - no need for the use of Optional here

File: src/main/java/uk/gov/defra/datareturns/validation/activities/ActivityValidator.java [hunk 2]
```diff
@@ -79,17 +82,30 @@ private boolean checkUniqueRiverPerSubmission(final Activity activity, final Con
...
+            if (valid && sub.get().getSource().equals(SubmissionSource.WEB)
+                    && activity.getDaysFishedWithMandatoryRelease() < 1 && activity.getDaysFishedOther() < 1) {
+                valid = handleError(context, "DAYS_FISHED_NOT_GREATER_THAN_ZERO", ConstraintValidatorContext.ConstraintViolationBuilder::addBeanNode);
+            }
+
+            if (valid && sub.get().getSource().equals(SubmissionSource.PAPER)
+                    && activity.getDaysFishedWithMandatoryRelease() < 0 && activity.getDaysFishedOther() < 0) {
+                valid = handleError(context, "DAYS_FISHED_LESS_THAN_ZERO", ConstraintValidatorContext.ConstraintViolationBuilder::addBeanNode);
+            }
```
Comment (sgardnerdell-ea) [inline src/main/java/uk/gov/defra/datareturns/validation/activities/ActivityValidator.java]: Need to add a test which checks that an appropriate error is generated for paper returns with negative effort values.

Is there a corresponding PR on the frontend for the change of validation message?

Comment (graham-willis-druid) [reply src/main/java/uk/gov/defra/datareturns/validation/activities/ActivityValidator.java]: The front end will just show "Enter a number between 0 and 167/198" if you add a negative effort 
The 0 is allowed in one box but not both for anglers of both for FMT.
